src/main/scala/no/ndla/draftapi/service/WriteService.scala

```diff
@@ -93,6 +94,8 @@ trait WriteService {
 
     def updateArticle(articleId: Long, updatedApiArticle: api.UpdatedArticle): Try[api.Article] = {
       draftRepository.withId(articleId) match {
+        case Some(existing) if existing.status.contains(QUEUED_FOR_PUBLISHING) =>
+          Failure(new AccessDeniedException("This article is marked for publishing and it cannot be updated until it is published"))
```

Comment:
Synes kanskje `AccessDeniedException` er litt rart å kaste her...? For meg betyr det at den som gjør en handling ikke har rettighet til det (f.eks. at man ikke har `publish`-rollen). Mens problemet her er at entiteten har feil status. Ville heller hatt noe `OperationNotAllowedException` el.

Reply:
...og så skal det vel være lov å oppdatere dersom du har publiserings-rettighet?

Reply:
Kan gjerne bytte til `OperationNotAllowedException`. Tenkte å beholde logikken så enkel som mulig i den første iterasjonen. Dersom en med publiseringsrollen skal få lov til å endre på en artikkel som er "til publisering" bør vel egentlig den statusen fjernes igjen. Det høres kanskje ut som noe vi håndterer i versjon 2?

Reply:
Mm. Enig i å ha det enkelt, men jeg skiller på det og "feil", som jeg mener dette er. At vi har feil exception der ville vi lett ha glemt i senere iterasjoner.

Godt spørsmål ang. tilgang. Jeg tenker at en med publiserings-rettighet får lov til å endre helt opptil publisering uten at den blir "tatt ut" av publiseringskøen av den grunn (og dermed åpner for alle andre å editere også). Men det må jo kunden bestemme til syvende og sist. 
Jeg tror bare at vi er nødt til å ha en mulighet for å gjøre noe med den "mens vi venter på batchjobben". Hvis ikke, tror jeg ikke det føles som en brukbar løsning...

Reply:
Ok. Da legger jeg inn støtte for at en med publiseringsrettigheter får lov til å oppdatere en artikkel når den står som publisert. Tenker da at "til publisering" statusen blir stående

Reply:
>...står som publisert
Du mener som `til publisering`? Etter at den er publisert får alle lov til å endre på den i draft-api (og dermed jobbe på en ny versjon)

Reply:
Ja. Skulle stå "til publisering"